PowershellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingService.cs

```diff
@@ -269,6 +269,15 @@ public Collection<Variable> GetScopedVariable()
 
                     if (psVar != null)
                     {
+                        if (psVar.Value is PSObject &&
+                            !(((PSObject)psVar.Value).ImmediateBaseObject is PSCustomObject))
+                        {

```

Comment:
Good question, actually I guess the ImmediateBaseObject's definition should explain it: "Gets the encapsulated base object. If the base object is another PSObject object, then its base object is returned."